src/v/serde/serde.h

```diff
@@ -589,6 +594,10 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
         auto second = read_nested<typename Type::second_type>(
           in, bytes_left_limit);
         t = std::make_pair(std::move(first), std::move(second));
+    } else if constexpr (is_duration_v<Type>) {
```

Comment:
I'd rather we were more explicit here. We have a few durations floating around in the codebase. Just recently seastar updated an underlying duration type. Without being very careful, this could easily break backwards compatibility if a duration type changes without the compiler complaining.

Reply:
Yikes, thats scary for sure, thanks for the context. I used the same trick as the one in  #5051 to serialize it as ns (for precision instead of ms). Let me know what you think.